src/axom/core/ArrayView.hpp

```diff
@@ -149,7 +149,7 @@ ArrayView<T, DIM, SPACE>::ArrayView(T* data, Args... args)
 {
   static_assert(sizeof...(Args) == DIM,
                 "Array size must match number of dimensions");
-#ifdef AXOM_DEVICE_CODE
+#if defined(AXOM_DEVICE_CODE) && defined(AXOM_USE_UMPIRE)
```

Comment:
Since my configuration did not have RAJA, I don't think `AXOM_DEVICE_CODE` should have been defined.
I think this is related to #734

Reply:
Weird...is this define (or `__CUDA_ARCH__`) coming from the XL compiler?

Reply:
Not sure -- I would expect it to be off when not in device code, and we shouldn't be in device code when we're not configured with RAJA.

In any event, `MemorySpace::Constant` is only defined when `AXOM_USE_UMPIRE`, so the extra guard shouldn't hurt.

Reply:
__CUDA_ARCH__ is only defined by nvcc when in device code. it is undefined in host code, which nvcc passes back to the host compiler so it can't be coming from XL. Is something amiss in our CMake setup that is not defining AXOM_DEVICE_CODE properly?